src/languages/asciidoc.js

```diff
@@ -110,30 +141,47 @@ export default function(hljs) {
       // inline unconstrained strong (single line)
       {
         className: 'strong',
-        begin: /\*\*(.+?)\*\*/
+        begin: `\\*{2}(${CC_ANY}+?)\\*{2}`
       },
       // inline unconstrained strong (multi-line)
       {
         className: 'strong',
-        begin: /\*{2}(\s\S+)*\*{2}/,
+        begin: /\*{2}(([^*\n\\]|\\[^\n])+\n)+([^*\n\\]|\\[^\n])*\*{2}/,
         relevance: 0
       },
-      // escaped formatting marks (i.e., \* \_ or \`)
+      // escaped constrained formatting marks (i.e., \* \_ or \`)
       {
         begin: /\\[*_`]/
       },
-      // inline contrained string (single line)
+      // escaped unconstrained formatting marks (i.e., \\** \\__ or \\``)
+      // must ignore until the next formatting marks
+      // this rule is not 100% compliant with Asciidoctor but we are entering undefined behavior territory...
+      {
+        begin: /\\\\\*{2}[^\n]*\*{2}/
+      },
+      {
+        begin: /\\\\_{2}[^\n]*_{2}/
+      },
+      {
+        begin: /\\\\`{2}[^\n]*`{2}/
+      },
+      // constrained preceded by ":", ";", or "}".
+      {
+        begin: /[:;}][*_`](?!\*)/
```

Comment:
Remind me what this is doing again? Is this a false positive guard or?

Reply:
Yes, we do not allow a constrained formatting mark to be preceded by ":", ";" or "}". I've added the following test cases to make sure that bold/strong is not applied:

```
There's a colon:*directly* before the starting formatting mark.

There's a semi-colon directly before the starting formatting mark &ndash;*2018*

There's a closing curly bracket directly {before}*the starting formatting mark*.
```